### PR TITLE
feat(deployment)!: use s3 native notifications over cloudtrail logging

### PR DESCRIPTION
**Pull request by saefty**
## what

Use s3 native logging over cloudtrail logging as it is cleaner and more scalable regardless of some central external config

## why

Recommended setup from AWS which is also simpler to maintain and more cost efficient, see https://docs.aws.amazon.com/codepipeline/latest/userguide/update-change-detection.html for details

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ module "lambda" {
 
       // optionally overwrite arguments like 'description'
       // from https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_event_rule
-      description = "Triggered by CloudTrail"
+      description = "Triggered by AWS s3 Event Notification"
```
Comment (moritzzimmer) [inline README.md:91]: This is unrelated and should not be part of this PR

File: README.md [hunk 2]
```diff
@@ -101,7 +101,7 @@ module "lambda" {
       event_pattern = <<PATTERN
       {
         "detail-type": [
-          "AWS Console Sign In via CloudTrail"
```
Comment (moritzzimmer) [inline README.md:104]: This is unrelated and should not be part of this PR

File: examples/with-cloudwatch-event-rules/main.tf [hunk 3]
```diff
@@ -23,7 +23,7 @@ module "lambda" {
 
       // optionally overwrite arguments like 'description'
       // from https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_event_rule
-      description = "Triggered by CloudTrail"
```
Comment (moritzzimmer) [inline examples/with-cloudwatch-event-rules/main.tf:26]: This is unrelated and should not be part of this PR

File: examples/with-cloudwatch-event-rules/main.tf [hunk 4]
```diff
@@ -36,7 +36,7 @@ module "lambda" {
       event_pattern = <<PATTERN
       {
         "detail-type": [
-          "AWS Console Sign In via CloudTrail"
```
Comment (moritzzimmer) [inline examples/with-cloudwatch-event-rules/main.tf:39]: This is unrelated and should not be part of this PR

File: modules/deployment/versions.tf [hunk 5]
```diff
@@ -2,6 +2,9 @@ terraform {
   required_version = ">= 0.12.0"
 
   required_providers {
-    aws = ">= 4.51"
+    aws = {
```
Comment (moritzzimmer) [inline modules/deployment/versions.tf:5]: IMO we should enforce TF >= 1.0 in a separate PR and use the new notation

File: versions.tf [hunk 6]
```diff
@@ -2,6 +2,9 @@ terraform {
   required_version = ">= 0.12.0"
 
   required_providers {
-    aws = ">= 4.51"
```
Comment (moritzzimmer) [inline versions.tf:5]: IMO we should enforce TF >= 1.0 in a separate PR and use the new notation

**Review by moritzzimmer — CHANGES_REQUESTED** [review]

File: .tflint.hcl [hunk 7]
```diff
@@ -4,7 +4,7 @@ config {
 
 plugin "aws" {
   enabled = true
-  version = "0.21.1"
+  version = "0.27.0"
```
Comment (moritzzimmer) [inline .tflint.hcl:7]: This is unrelated and should not be part of this PR

**Review by moritzzimmer — CHANGES_REQUESTED** [review]

File: modules/deployment/versions.tf (hunk 5 above)
Comment (saefty) [reply modules/deployment/versions.tf:5]: The pipelines fail though if I dont add the new notation due to tflint detecting this as an issue

**Review by moritzzimmer — APPROVED** [review]

File: modules/deployment/main.tf [hunk 8]
```diff
@@ -130,6 +130,13 @@ resource "aws_s3_bucket" "pipeline" {
   tags          = var.tags
 }
 
+resource "aws_s3_bucket_notification" "pipeline" {
```
Comment (moritzzimmer) [inline modules/deployment/main.tf:133]: guess we won't need this resource, since we need notifications from the source bucket

**Review by moritzzimmer — CHANGES_REQUESTED** [review]

**Review by moritzzimmer — APPROVED** [review]
